Tests/MongoSwiftTests/ReadPreferenceTests.swift

```diff
@@ -8,7 +8,7 @@ final class ReadPreferenceTests: MongoSwiftTestCase {
     }
 
     func testMode() {
-        let primary = ReadPreference(.primary)
```

Comment:
I think instead of replacing this case, we should just add a new one verifying the default behavior.